snap/local/build-helpers/include/videodev2.h

```diff
@@ -0,0 +1,2581 @@
+/* SPDX-License-Identifier: ((GPL-2.0+ WITH Linux-syscall-note) OR BSD-3-Clause) */
+/*
+ *  Video for Linux Two header file
```

Comment:
Why are we copying a Linux kernel header file into the snap build-helpers directory?

```
dpkg -S videodev2.h
linux-hwe-5.13-headers-5.13.0-40: /usr/src/linux-hwe-5.13-headers-5.13.0-40/include/uapi/linux/videodev2.h
linux-hwe-5.13-headers-5.13.0-41: /usr/src/linux-hwe-5.13-headers-5.13.0-41/include/uapi/linux/videodev2.h
linux-hwe-5.11-headers-5.11.0-34: /usr/src/linux-hwe-5.11-headers-5.11.0-34/include/uapi/linux/videodev2.h
linux-hwe-5.13-headers-5.13.0-40: /usr/src/linux-hwe-5.13-headers-5.13.0-40/include/linux/videodev2.h
linux-libc-dev:amd64: /usr/include/linux/videodev2.h
linux-hwe-5.13-headers-5.13.0-41: /usr/src/linux-hwe-5.13-headers-5.13.0-41/include/linux/videodev2.h
linux-hwe-5.11-headers-5.11.0-34: /usr/src/linux-hwe-5.11-headers-5.11.0-34/include/linux/videodev2.h
linux-hwe-5.8-headers-5.8.0-63: /usr/src/linux-hwe-5.8-headers-5.8.0-63/include/linux/videodev2.h
linux-hwe-5.8-headers-5.8.0-63: /usr/src/linux-hwe-5.8-headers-5.8.0-63/include/uapi/linux/videodev2.h
```

**Note** - the above output also shows this header included the dev package for glibc. I honestly don't know enough about v4l2 to say why.

In general, it's not a good idea to copy header files from other projects. If you do, then it's also best practice to be very clear about where the file was sourced from and why. This file is also dual licensed (GPLv2 or BSD), and has some terms about copyright notices that must be included in both **source** and **binary** distributions.

I also ran a diff of a few of the different versions of the header file and they're not all the same across different kernel versions.

This also brings up questions as to whether this device service is agnostic to the underlying kernel version?

Reply:
I've explained in the snapcraft file why the header is needed.

Good point about the license. However, the header file is included in the resulting snap whether or not I package this version of it. The snap is simply replacing the existing one with a copy that has the set of definitions needed for the compilation.

Regarding being agnostic to the underlying kernel, this probably an issue from the dependency: https://github.com/vladimirvivien/go4vl#compilation-requirements

Reply:
See issues #37